app/src/main/java/de/danoeh/antennapod/menuhandler/FeedItemMenuHandler.java

```diff
@@ -135,6 +143,40 @@ public static boolean onPrepareMenu(Menu menu, FeedItem selectedItem, int... exc
         return rc;
     }
 
+    public static boolean onMenuItemClicked(@NonNull Fragment fragment, int menuItemId,
+                                            @NonNull Feed selectedFeed) {
+
+        @NonNull Context context = fragment.requireContext();
+        if (menuItemId == R.id.rename_folder_item) {
+            new RenameItemDialog(fragment.getActivity(), selectedFeed).show();
+        } else if (menuItemId == R.id.remove_all_inbox_item) {
+            DisplayConfirmationDialog.display(fragment.getActivity(), R.string.remove_all_inbox_label,
```

Comment:
I don't think `DisplayConfirmationDialog` is really necessary. I would just move the same code structure that was before used in `SubscriptionFragment` instead of encapsulating the `ConfirmationDialog` in an additional layer. (now that menu clicks are handled in one single location, there won't be any code duplication)